hydroflow_macro/build.rs

```diff
@@ -79,44 +80,62 @@ fn main() -> Result<(), Box<dyn Error>> {
-                    write,
-                    "> Input port names: {}  ",
-                    port_names
-                        .into_iter()
-                        .map(|idx| format!("`{}`, ", idx.into_token_stream()))
-                        .collect::<String>()
-                        .strip_suffix(", ")
-                        .unwrap_or("&lt;EMPTY&gt;")
-                )?,
-                PortListSpec::Variadic => writeln!(
-                    write,
-                    "> Input port names: Variadic, as specified in arguments."
-                )?,
-            }
+            input_str_maybe = Some(match (f)() {
+                PortListSpec::Fixed(port_names) => {
+                    port_names.iter().for_each(|port| {
+                        let port_ix = match port.clone() {
+                            PortIndex::Int(x) => PortIndexValue::Int(x),
+                            PortIndex::Path(x) => PortIndexValue::Path(x),
+                        };
+                        if let Some(DelayType::Stratum) = (op.input_delaytype_fn)(&port_ix) {
+                            blocking = true;
+                        }
+                    });
+                    format!(
+                        "> Input port names: {}  ",
+                        port_names
+                            .into_iter()
+                            .map(|idx| format!("`{}`, ", idx.into_token_stream()))
+                            .collect::<String>()
+                            .strip_suffix(", ")
+                            .unwrap_or("&lt;EMPTY&gt;")
+                    )
+                }
+                PortListSpec::Variadic => {
+                    format!("> Input port names: Variadic, as specified in arguments.")
+                }
+            })
         }
+        let mut output_str_maybe = None;
         if let Some(f) = op.ports_out {
-            match (f)() {
-                PortListSpec::Fixed(port_names) => writeln!(
-                    write,
+            output_str_maybe = Some(match (f)() {
+                PortListSpec::Fixed(port_names) => format!(
                     "> Output port names: {}  ",
                     port_names
                         .into_iter()
                         .map(|idx| format!("`{}`, ", idx.into_token_stream()))
                         .collect::<String>()
                         .strip_suffix(", ")
                         .unwrap_or("&lt;EMPTY&gt;")
-                )?,
-                PortListSpec::Variadic => writeln!(
-                    write,
-                    "> Output port names: Variadic, as specified in arguments."
-                )?,
-            }
+                ),
+                PortListSpec::Variadic => {
+                    format!("> Output port names: Variadic, as specified in arguments.")
+                }
+            })
+        }
+        let flow_str = if blocking { "Blocking" } else { "Streaming" };
```

Comment:
Nothings fully "blocking", that doesnt match what `DelayType` means, it's more what kind of barrier it is.

It might also make sense to specify *which* input has the barrier

Reply:
or "batches per-stratum"/"delays for an epoch"